blockchain/chain.go

```diff
@@ -667,6 +671,37 @@ func (b *BlockChain) connectBlock(node *blockNode, block *btcutil.Block,
 	b.sendNotification(NTBlockConnected, block)
 	b.chainLock.Lock()
 
+	// If the pruneTarget isn't 0, we should attempt to delete older blocks
+	// from the database.
+	if b.pruneTarget != 0 {
+		err = b.db.Update(func(dbTx database.Tx) error {
+			// When the total block size is under the prune target, prune blocks is
+			// a no-op and the deleted hashes are nil.
+			deletedHashes, err := dbTx.PruneBlocks(b.pruneTarget)
```

Comment:
Why not run this in the above transaction where all the other indexes are updated?